src/components/Admin/ManageEvents.js

```diff
@@ -0,0 +1,734 @@
+                Clear
+              </Button>
+            )}
+          </FullWidthFormGroup>
+        </FullWidthFormRow>
+      )}
+
+      <SubmitButton />
+    </CenteredForm>
+  );
+};
+
+class ManageEvents extends Component {
+  _initFirebase = false;
+  state = {
+    edit: false,
+    loading: true,
+    maxIndex: 1000000,
+    events: [],
+    currentEvent: null,
+    currentEventWebsite: null,
+    showModal: false,
+    showQRModal: false,
+    modalEvent: DEFAULT_EVENT,
+    modalWebsiteEvent: DEFAULT_WEBSITE_EVENT,
+  };
+  unsubEvents = null;
+  unsubCurrentEventWebsite = null;
+
+  static contextType = AuthUserContext;
+
+  componentDidMount() {
+    if (this.props.firebase && !this._initFirebase) this.loadData();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.firebase && !this._initFirebase) this.loadData();
+  }
+
+  componentWillUnmount() {
+    if (typeof this.unsubEvents === "function") this.unsubEvents();
+    if (typeof this.unsubCurrentEventWebsite === "function")
+      this.unsubCurrentEventWebsite();
+  }
+
+  loadData = async () => {
+    this._initFirebase = true;
+
+    await this.props.firebase.getIndex().then((doc) => {
+      this.setState({ maxIndex: doc.data().maxIndex + 1 });
+    });
+
+    const events = await new Promise((resolve, reject) => {
+      let resolveOnce = (doc) => {
+        resolveOnce = () => null;
+        resolve(doc);
+      };
+      this.unsubEvents = this.props.firebase
+        .recruitmentEvents()
+        .onSnapshot((querySnapshot) => {
+          const events = querySnapshot.docs.map((doc) => {
+            return {
+              ...doc.data(),
+              uid: doc.id,
+            };
+          });
+
+          this.setState({ events });
+          resolveOnce(events);
+        }, reject);
+    });
+
+    this.setState({
+      events,
+      loading: false,
+    });
+  };
+
+  fetchEvent = (uid) => {
+    if (!uid) return;
+
+    if (typeof this.unsubCurrentEventWebsite === "function")
+      this.unsubCurrentEventWebsite();
+    this.unsubCurrentEventWebsite = this.props.firebase
+      .event(uid)
+      .onSnapshot((doc) => {
+        if (!doc.exists) return console.log("Event not found!");
+        const fetchedEvent = { ...doc.data(), uid: doc.id };
+        this.setState({ currentEventWebsite: fetchedEvent });
+      });
+  };
+
+  toggleEventStatus = async () => {
+    const { currentEvent, currentEventWebsite } = this.state;
+    if (!currentEventWebsite || !currentEvent) return;
+
+    const updatedEvent = cloneDeep(currentEvent);
+    updatedEvent.open = !updatedEvent.open;
+
+    delete updatedEvent.uid;
+
+    await this.props.firebase
+      .recruitmentEvent(currentEvent.uid)
+      .set(updatedEvent)
+      .then(() => {
+        updatedEvent.uid = currentEvent.uid;
+        this.setState({ currentEvent: updatedEvent });
+        this.fetchEvent(updatedEvent.uid);
+      })
+      .catch((error) => console.log("Failed to save event"));
```

Comment:
I'm not 100% sure how this would work, do I just leave the error without a catch and the boundary will do its job?

Reply:
Hmm so I'm rethinking my original comment. Error boundaries should be placed such that they logically encapsulate some portion of the app, e.g. say for some reason an error was preventing an event from loading correctly, the error boundary would then show a message instead of the event with broken data. In this case, where we're making an API call that fails, I'm not sure if an error boundary is the right way to display the error here. For a failed API call I think the most intuitive thing would a toast/alert of some kind that tells the user what went wrong (feel free to still console.error so you have the data there, but the user should get a simplified message so they're not wondering why nothing happened)